### PR TITLE
APP-172: GPIO app was moved from ant to gradle and updated

### PR DESCRIPTION
**Pull request by iamtodor**

### CODE REVIEW

**Comment by Slash32** [comment]
@kaa-leeroy test this please

File: gpiocontrol/source/android/app/src/main/java/org/kaaproject/kaa/examples/gpiocontol/utils/KaaProvider.java [hunk 1]
```diff
@@ -25,8 +28,6 @@
 import org.kaaproject.kaa.common.endpoint.gen.UserAttachResponse;
 import org.kaaproject.kaa.examples.gpiocontrol.DeviceInfoRequest;
 import org.kaaproject.kaa.examples.gpiocontrol.RemoteControlECF;
-import android.content.Context;
-import android.util.Log;
 
 public class KaaProvider {
```
Comment (maxml) [inline gpiocontrol/source/android/app/src/main/java/org/kaaproject/kaa/examples/gpiocontol/utils/KaaProvider.java]: I think this class must be named KaaManager like in other apps and must be in directory kaa/ like in other apps.

File: gpiocontrol/source/android/app/src/main/java/org/kaaproject/kaa/examples/gpiocontol/utils/PreferencesManager.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ */
+
+package org.kaaproject.kaa.examples.gpiocontol.utils;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+
+public class PreferencesManager {
+
```
Comment (maxml) [inline gpiocontrol/source/android/app/src/main/java/org/kaaproject/kaa/examples/gpiocontol/utils/PreferencesManager.java:24]: I think, this app must be the simplest. New class for three lines - it's not unnecessary.

File: gpiocontrol/source/android/app/src/main/java/org/kaaproject/kaa/examples/gpiocontol/model/Device.java [hunk 3]
```diff
@@ -42,22 +42,30 @@ public Device(String model, String deviceName, List<GpioStatus> gpioStatusList,
         this.kaaEndpointId = kaaEndpointId;
     }
 
-    public String getModel() {
-        return model;
+    @Override
```
Comment (maxml) [inline gpiocontrol/source/android/app/src/main/java/org/kaaproject/kaa/examples/gpiocontol/model/Device.java:45]: In This class please move methos to some order - public methods, Override, private

File: gpiocontrol/source/android/app/src/main/java/org/kaaproject/kaa/examples/gpiocontol/DevicesListActivity.java [hunk 4]
```diff
@@ -1,41 +1,39 @@
 /**
```
Comment (maxml) [inline gpiocontrol/source/android/app/src/main/java/org/kaaproject/kaa/examples/gpiocontol/DevicesListActivity.java]: please, move all activities to some directory, for example ui/ Or you can use my approach and leave only one and main activity, activity which app loads. But two activities can't be in root directory

File: gpiocontrol/source/android/app/src/main/java/org/kaaproject/kaa/examples/gpiocontol/DevicesListActivity.java [hunk 5]
```diff
@@ -111,29 +104,34 @@ private void addDevice(Device device) {
...
-                PreferencesManager.setUserExternalId(this, "2");
-                Log.d(TAG, "Attaching user...");
-                KaaProvider.attachUser(context);
-            }
-            setUpEndpointListener();
+
+        kaaClient = KaaProvider.getClient(this);
+        kaaClient.start();
+        if (isFirstLaunch()) {
+            PreferencesManager.setUserExternalId(this, "2");
```
Comment (maxml) [inline gpiocontrol/source/android/app/src/main/java/org/kaaproject/kaa/examples/gpiocontol/DevicesListActivity.java]: Why 2? Why not in constant?

File: gpiocontrol/source/android/app/src/main/java/org/kaaproject/kaa/examples/gpiocontol/DevicesListActivity.java [hunk 6]
```diff
@@ -57,18 +56,13 @@
 
     private final String TAG = DevicesListActivity.class.getSimpleName();
```
Comment (maxml) [inline gpiocontrol/source/android/app/src/main/java/org/kaaproject/kaa/examples/gpiocontol/DevicesListActivity.java]: Please, provide some comments to this class. What does he do? Simlar to other classes - please, add comments

File: gpiocontrol/source/android/app/src/main/java/org/kaaproject/kaa/examples/gpiocontol/DevicesListActivity.java [hunk 7]
```diff
@@ -190,30 +193,26 @@ public void onAttach(SyncResponseResultType syncResponseResultType, EndpointKeyH
...
+                        })
+                .show();
     }
 
-    private boolean isFirstLaunch(){
+    private boolean isFirstLaunch() {
         return PreferencesManager.getUserExternalId(this).isEmpty();
     }
 
     private void initViews() {
```
Comment (maxml) [inline gpiocontrol/source/android/app/src/main/java/org/kaaproject/kaa/examples/gpiocontol/DevicesListActivity.java]: why use separate method?

**Review by Slash32 — CHANGES_REQUESTED** [review]
Please address the comments to the PR prior to merge.

**Comment by Slash32** [comment]
@kaa-leeroy test this please

**Comment by Slash32** [comment]
@kaa-leeroy test this please

File: gpiocontrol/source/android/app/src/main/java/org/kaaproject/kaa/examples/gpiocontol/utils/KaaProvider.java (hunk 1 above)
Comment (iamtodor) [reply gpiocontrol/source/android/app/src/main/java/org/kaaproject/kaa/examples/gpiocontol/utils/KaaProvider.java]: Done

File: gpiocontrol/source/android/app/src/main/java/org/kaaproject/kaa/examples/gpiocontol/utils/PreferencesManager.java (hunk 2 above)
Comment (iamtodor) [reply gpiocontrol/source/android/app/src/main/java/org/kaaproject/kaa/examples/gpiocontol/utils/PreferencesManager.java:24]: Don't agree

File: gpiocontrol/source/android/app/src/main/java/org/kaaproject/kaa/examples/gpiocontol/model/Device.java (hunk 3 above)
Comment (iamtodor) [reply gpiocontrol/source/android/app/src/main/java/org/kaaproject/kaa/examples/gpiocontol/model/Device.java:45]: Done

File: gpiocontrol/source/android/app/src/main/java/org/kaaproject/kaa/examples/gpiocontol/DevicesListActivity.java (hunk 4 above)
Comment (iamtodor) [reply gpiocontrol/source/android/app/src/main/java/org/kaaproject/kaa/examples/gpiocontol/DevicesListActivity.java]: Done

File: gpiocontrol/source/android/app/src/main/java/org/kaaproject/kaa/examples/gpiocontol/DevicesListActivity.java (hunk 5 above)
Comment (iamtodor) [reply gpiocontrol/source/android/app/src/main/java/org/kaaproject/kaa/examples/gpiocontol/DevicesListActivity.java]: Have no idea what is it. I turned it to the constant.

**Comment by iamtodor** [comment]
@maxml pls, review

**Review by Slash32 — APPROVED** [review]

**Pull request merged**